helm: configure dynamoDB autoscaling in teleport-cluster (#13808)
gravitational/teleport: docs/pages/setup/helm-deployments/aws.mdx

  To change `chartMode`, `clusterName`, or any `aws` settings, you must first uninstall the existing chart and then install a new version with the appropriate values.
</Admonition>

### Autoscaling

In order to reduce DynamoDB costs you might want to enable DynamoDB autoscaling.
This step is usually done after a successful Teleport deployment, once you have gathered some data
about Teleport's DynamoDB usage and know what regular usage looks like and how autoscaling should be tuned.

> You must know the desired read/write minimum, maximum and target capacity.

Reviewer: ```suggestion
You must know the desired read/write minimum, maximum and target capacity for your DynamoDB instance in order to enable autoscaling.
```

> Autoscaling configuration can be delegated to Teleport or managed by an AWS Application Auto Scaling policy you create.

Reviewer: ```suggestion
You can delegate your autoscaling configuration to Teleport or manage it by creating an AWS Application Auto Scaling policy.
```

  To change `chartMode`, `clusterName`, or any `aws` settings, you must first uninstall the existing chart and then install a new version with the appropriate values.
</Admonition>

### Autoscaling

In order to reduce DynamoDB costs you might want to enable DynamoDB autoscaling.
This can be done manually or delegated to Teleport.

> This is step is usually done after a successful Teleport deployment, once you have gathered some data

Reviewer: ```suggestion
This step is usually done after a successful Teleport deployment, once you have gathered some data
```

about Teleport's DynamoDB usage and know what regular usage looks like and how autoscaling should be tuned.
See [the DynamoDB autoscaling section](../reference/backends.mdx#dynamodb-autoscaling) for more details.

> To go further you have to know the desired read/write minimum, maximum and target capacity.

Reviewer: What does "To go further" mean here? Can we combine this sentence with the previous paragraph, which already says that autoscaling is usually set up after gathering some data?

If you decide to let Teleport set up the autoscaling, proceed as follows:

<Tabs>
  <TabItem label="Using values.yaml">
  Set the following fields in your existing `aws-values.yaml` file and replace the numeric values with yours:

  ```yaml
  aws:
    # [...] already present values under `aws`
    dynamoAutoScaling: true
    readMinCapacity: 5     # integer
    readMaxCapacity: 100   # integer
    readTargetValue: 50.0  # float
    writeMinCapacity: 5    # integer
    writeMaxCapacity: 100  # integer
    writeTargetValue: 50.0 # float

> 

Reviewer: This needs a closing triple-backtick

Author: Fixed

>   Then perform [a cluster upgrade](#upgrading-the-cluster-after-deployment) with the new values.

Reviewer: If this is only the `helm upgrade` command, I think we can just include the command snippet here as well (rather than use a link) so the reader can see all of these instructions in one place.

Author: Done

  </TabItem>

>   <TabItem label="Using --set via CLI">

Reviewer: Do we need this `TabItem`? There are a lot of `--set` flags here, so I'm not sure there would be a case where `--set` would be more convenient than using a values file.

Author: Removed
